charts/newrelic-infrastructure/templates/deployment.yaml

```diff
@@ -0,0 +1,241 @@
+            {{- end }}
+            {{- end }}
+            {{- if .Values.customAttributes }}
+            - name: "NRIA_CUSTOM_ATTRIBUTES"
+              value: {{ .Values.customAttributes }}
+            {{- end }}
+            - name: "NRIA_PASSTHROUGH_ENVIRONMENT"
+              value: "KUBERNETES_SERVICE_HOST,KUBERNETES_SERVICE_PORT,CLUSTER_NAME,CADVISOR_PORT,NRK8S_NODE_NAME,KUBE_STATE_METRICS_URL,KUBE_STATE_METRICS_POD_LABEL,TIMEOUT,ETCD_TLS_SECRET_NAME,ETCD_TLS_SECRET_NAMESPACE,API_SERVER_SECURE_PORT,KUBE_STATE_METRICS_SCHEME,KUBE_STATE_METRICS_PORT,KUBE_STATE_METRICS_NAMESPACE,SCHEDULER_ENDPOINT_URL,ETCD_ENDPOINT_URL,CONTROLLER_MANAGER_ENDPOINT_URL,API_SERVER_ENDPOINT_URL,DISCOVERY_CACHE_TTL,DISABLE_KUBELET"
+            {{- if .Values.verboseLog }}
+            - name: NRIA_VERBOSE
+              value: "1"
+            {{- end }}
+            {{- if .Values.logFile }}
+            - name: NRIA_LOG_FILE
+              value: {{ .Values.logFile }}
+            {{- end }}
+            # Using FORWARD_ONLY mode to avoid the entity creation for the events.
+            - name: NRIA_IS_SECURE_FORWARD_ONLY
+              value: "false"
+            - name: NRIA_IS_FORWARD_ONLY
+              value: "true"
+          volumeMounts:
+            {{- if .Values.config }}
+            - name: config
+              mountPath: /etc/newrelic-infra.yml
+              subPath: newrelic-infra.yml
+            {{- end }}
+            {{- if .Values.integrations_config }}
+            - name: nri-integrations-cfg-volume
+              mountPath: /etc/newrelic-infra/integrations.d/
+            {{- end }}
+            - mountPath: /var/db/newrelic-infra/data
+              name: tmpfs-data
+            - mountPath: /var/db/newrelic-infra/user_data
+              name: tmpfs-user-data
+            - mountPath: /tmp
+              name: tmpfs-tmp
+            - mountPath: /var/cache/nr-kubernetes
+              name: tmpfs-cache
+          {{- if .Values.resources }}
+          resources:
+{{ toYaml .Values.resources | indent 12 }}
+          {{- end }}
+      volumes:
+        - name: tmpfs-data
+          emptyDir: {}
+        - name: tmpfs-user-data
+          emptyDir: {}
+        - name: tmpfs-tmp
+          emptyDir: {}
+        - name: tmpfs-cache
+          emptyDir: {}
+        {{- if .Values.config }}
+        - name: config
+          configMap:
+            name: {{ template "newrelic.fullname" . }}
+            items:
+            - key: newrelic-infra.yml
+              path: newrelic-infra.yml
+        {{- end }}
+        {{- if .Values.integrations_config }}
+        - name: nri-integrations-cfg-volume
+          configMap:
+            name: {{ template "newrelic.fullname" . }}-integrations-cfg
+        {{- end }}
+      {{- if $.Values.priorityClassName }}
+      priorityClassName: {{ $.Values.priorityClassName }}
+      {{- end }}
+      affinity:
+        podAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
```

Comment:
Definitely out of scope for the PoC but I'd add the reciprocal label to KSM, so the affinity is honored if the KSM pod gets killed.

Reply:
True. I mean, this is only to make it work on cluster with multiple nodes in best case scenario :smile: But we can consider adding those affinities in general.